### PR TITLE
met variables

### PR DESCRIPTION
**Pull request by zdemirag**
Met variables are changed to 4 vectors. met up to eta 3.0 is calculated. 
extendMet variable is implemented to activate / deactivate the mentioned branches.

bambu implementation is not there yet.

### CODE REVIEW

**Comment by yiiyama** [comment]
Let me know if you want me to write the Bambu part.

**Comment by zdemirag** [comment]
I was just taking a look, I guess there needs to be a new implementation of newP4. Is that correct?
Feel free to implement the bambu if you have time, I am looking into something else right now.. which will take an other hour or so. Otherwise I was planning on playing around a new implementation of newP4 function.

**Comment by yiiyama** [comment]
Not quite, since your newly added TLorentzVectors are not in TClonesArray. newP4 was just a convenience function to not have to write the rather awkward "new (pointer) object" syntax.

**Comment by yiiyama** [comment]
I have the filler from bambu ready and can make a PR after this one is merged.
One minor comment: in the current version the newly added TLorentzVectors are not allocated by us but by ROOT when null pointers are passed in Branch() and SetBranchAddress(). There is nothing wrong about it; just pointing out that this behavior is different from other object branches of Bare objects.

File: Core/src/BareMet.cpp [hunk 1]
```diff
@@ -50,21 +49,19 @@ void BareMet::defineBranches(TTree *t){
...
-    //
-    t->Branch("metChargedHadron",&phiChargedHadron,"Phi/F");
-    //
-    t->Branch("metNeutralHadron",&metNeutralHadron,"Pt/F");
-    //
-    t->Branch("metNeutralHadron",&phiNeutralHadron,"Phi/F");
-    //
-    t->Branch("metNeutralEM",&metNeutralEM,"Pt/F");
+    if ( IsExtend() )
+    {
```
Comment (amarini) [inline Core/src/BareMet.cpp]: You miss the constructors (new ... ) here.

File: Core/src/BareMet.cpp [hunk 2]
```diff
@@ -78,14 +75,14 @@ void BareMet::setBranchAddresses(TTree *t){
...
-    t->SetBranchAddress("phiNoMu", &phiNoMu);
-    t->SetBranchAddress("metChargedHadron", &metChargedHadron);
-    t->SetBranchAddress("phiChargedHadron", &phiChargedHadron);
-    t->SetBranchAddress("metNeutralHadron", &metNeutralHadron);
-    t->SetBranchAddress("phiNeutralHadron", &phiNeutralHadron);
-    t->SetBranchAddress("metNeutralEM", &metNeutralEM);
-    t->SetBranchAddress("phiNeutralEM", &phiNeutralEM);
+    if ( IsExtend() ) 
+    {
+        t->SetBranchAddress("metNoMu", &metNoMu);
```
Comment (amarini) [inline Core/src/BareMet.cpp]: and here.

**Comment by amarini** [comment]
* You miss the constructor of the TLorentzVectors.
* Maybe we can align also the genMet as a TLorentzVector and not a TClonesArray ? (no particular feeling on it.)

**Pull request merged**